toolbox/compiler_finder.py

```diff
@@ -0,0 +1,112 @@
+#!/usr/bin/env python3
+#
+# Helper to check the satisfying compilers for C and C++
+#
+#########################################################################################
+#
+# CelestialCalendar Automation:
+#   Python automation scripts for building and testing the CelestialCalendar C++ project.
+# 
+# Author : Ningqi Wang (0xf3cd)
+# Email  : nq.maigre@gmail.com
+# Repo   : https://github.com/0xf3cd/celestial-calendar
+# License: GNU General Public License v3.0
+# 
+# This software is distributed without any warranty.
+# See <https://www.gnu.org/licenses/> for more details.
+
+import os
+import re
+import sys
+import shutil
+import argparse
+
+from pathlib import Path
+from typing import Dict, List
+
+# Apply a workaround to import from the parent directory...
+sys.path.append(str(Path(__file__).parent.parent))
+
+from automation import (
+  check_c_support, check_cpp_support, CompilerArgs, make_compiler_args,
+)
+
+
+def c_compilers(standards: List[str]) -> Dict[CompilerArgs, bool]:
+  '''Find the C compilers in `PATH`, and find out if they support the given standards.'''
+  c_compilers_pattern = re.compile(r'^(gcc|clang|icc)(\d*)$')
```

Comment:
Compilers like gcc14, clang-18, clang18 should also be supported. Same for C++ compilers.